### ISSUE TITLE
Proposal: Add a Writing Doc overview to the Contributing Guide

### ISSUE THREAD

**Issue by pushfoo** [pythonarcade/arcade#1886]
## Documentation request:

tl;dr add a Contributing Guide page / section containing the most common and/or arcade-specific sphinx concerns

This is a follow-up to a [Discord conversation](https://discord.com/channels/458662222697070613/705076586986078350/1143983620256694457)

### What the overview should have

* What, where, when, and how to add things:
  * Purpose & location of each:
    * Programming guide
    * Examples
    * Tutorials
    * Docstrings

 * Guidelines for instructions 
   * Link an external source if other general doc exists
   * Put it on its own page otherwise

* Which build formats are priority support targets
  * Model off of NetBSD's tier system? (https://www.netbsd.org/ports/#tiers)
  * Arcade doc support tiers prioritize game prototyping and offline education
    1. Primary: HTML readthedocs
    2. Secondary: PDF, ePUB (?)
    3. When it doesn't interfere with the top 2 + time allows: Everything else

* The most common problems and how to deal with them
  * Sphinx header underlines must have correct length to avoid sphinx warnings which break build

  * tl;dr of how to link things
    - [ ] Agree on link name pattern / structure
    * Internal links: <code>:ref:\`internal_link_name\`</code> 
    * Named external links need trailing underscore in the angle brackets:
      * Correct: <code>\`text <ref_name_>\`\_</code>
      * Broken: <code>\`text <ref_name>\`_</code>
    * Intersphinx usage:
      * For Python doc: <code>:py:class:\`Sequence\`</code> should just work
      * pyglet should be the same way, but may require some prefixes
      * Unsure if should be included: `:external:` for link targets to headings

  * Toctrees
    * Item not in any toctree 
    * @cspotcode's `#include` analogy
    *  tl;dr of what a virtual hierarchy is
    * `:hidden:`, `:includehidden:` & furo

* Tips, such as HTML-only behavior
  * rawhtml isn't rendered in PDFs
     - [ ] Find out whether this applies to other build formats we might want (ePUB, other ebook formats)

* Links to more general sphinx doc
  * Practical smaller wikis (ie https://lpn-doc-sphinx-primer.readthedocs.io/en/stable/index.html#)
  * Google style guide, etc
  * Relevant parts of general sphinx doc

### What the page should avoid

* Explanations of general Sphinx functionality when not directly relevant to arcade problems
* Detailed instructions

**Comment by pushfoo** [comment]
An optional item I'm not yet sure on: using [the `:external:` role with intersphinx](https://www.sphinx-doc.org/en/master/usage/extensions/intersphinx.html#role-external).

Pros:
1. If link target names stay the same, URLs auto-update on each build
2. Cleaner than including the entire URL
3. Does not break existing links

Cons:
1. Assumes the link target name will stay the same in the target project
2. Assumes we've added a project in intersphinx
3. Assumes projects will continue exporting `objects.inv` files

**Comment by pushfoo** [comment]
Further style notes from recent PR reviews / etc:

1. Use ``.. code-block:: python` instead of the `End of line::\n` syntax
2. Use the english word when writing small numbers < 10 (Contested, > 3 should be a number)
